doc/leak_tracking/MOTIVATION.md

```diff
@@ -0,0 +1,81 @@
+
+Coming soon! See https://github.com/flutter/devtools/issues/3951.
+
+The text below is under construction.
+
+# Motivation for leak tracking
+
+This page describes motivation behind memory leak tracking.
+Read more about leak tracking in [overview](OVERVIEW.md).
+
+## Why do disposables exist?
+
+In most cases object lifecycle is managed well by a garbage collector.
+
+However, some objects (memory-risky objects) require special care, to avoid risk of memory leaks:
+1. Objects that allocate significant memory for their members
+2. Objects that may reference a significant number of other objects, designed to be short living (like the entire widget tree), and prevent them from being GCed in time.
+
+![leak](images/leak.png "Leak")
+
+Such objects have method `dispose`, which has two purposes:
+
+1. Releases the resources, to make sure they do not stay longer than needed.
+2. Explicitly marks the object as not needed, to make lifecycle management easier. For example:
+    a. Some objects restrict invocation of some methods after disposal.
+    b. It is expected that objects should be available for GC after disposal, so owners of the objects should null the references after invocation of `dispose`.
+
+## Memory efficiency of applications
+
+To ensure **memory efficiency** of applications, engineers should watch that:
+1. All disposables are disposed when they are not needed any more.
+2. Large objects and objects, which reference big sets of other objects, are not reachable after usage.
```

Comment:
Large objects and objects that reference large sets of other objects are not reachable after disposal

Reply:
They not necessarily disposable.

Reply:
Okay. Then my suggestion is the same wording above but with "usage":

"Large objects and objects that reference large sets of other objects are not reachable after usage."